eva/executor/vector_index_scan_executor.py

```diff
@@ -62,10 +71,13 @@ def exec(self, *args, **kwargs) -> Iterator[Batch]:
         search_batch.drop_column_alias()
         search_feat = search_batch.column_as_numpy_array(feature_col_name)[0]
         search_feat = search_feat.reshape(1, -1)
-        dis_np, row_id_np = index.search(search_feat, self.limit_count.value)
-
+        index_result = self.index.query(
+            VectorIndexQuery(search_feat, self.limit_count.value)
+        )
+        dis_np = index_result.similarities
+        row_id_np = index_result.ids
         distance_list, row_id_list = [], []
-        for dis, row_id in zip(dis_np[0], row_id_np[0]):
+        for dis, row_id in zip(dis_np, row_id_np):
```

Comment:
Are we generalizing the array length here? Some comments in the code on what is happening will help.